src/tools/tests/allocator-recipe-resolver-test.ts

```diff
@@ -1149,6 +1157,167 @@ recipe ReadingRecipeAD
+      '[Thing]',
+      '[Thing {a: Text, b: Text, d: Text}]'
+    );
+    await verifyPhantomReader(
+      thingSchema,
+      'from Thing access {a, b, d}',
+      '&Thing',
+      '&Thing {a: Text, b: Text, d: Text}'
+    );
+    await verifyPhantomReader(
+      thingSchema,
+      'from Thing access {a, b, d}',
+      '[&Thing]',
+      '[&Thing {a: Text, b: Text, d: Text}]'
+    );
+    const thingSchemaWithLists = `
+      schema Thing
+        a: List<Text>
+        b: List<Text>
+        c: List<Text>
+        d: List<Text>
+        e: List<Text>
+    `;
+    await verifyPhantomReader(
+      thingSchemaWithLists,
+      'from Thing access {a, b, d}',
+      'Thing',
+      'Thing {a: List<Text>, b: List<Text>, d: List<Text>}'
+    );
+    await verifyPhantomReader(
+      thingSchemaWithLists,
+      'from Thing access {a, d}',
+      'Thing',
+      'Thing {a: List<Text>, d: List<Text>}'
+    );
+    // Inline entity tests
+    const thingSchemaWithInline = `
+      schema Thing
+        foo: inline Foo {
+          a: Text,
+          b: Text,
+          c: Text,
+          d: Text,
+          e: Text
+        }
+        bar: Text
+    `;
+    await verifyPhantomReader(
+      thingSchemaWithInline,
+      `from Thing access {
+         foo {a, b, d}
+      }`,
+      'Thing',
+      'Thing {foo: inline Foo {a: Text, b: Text, d: Text}}'
+    );
+    await verifyPhantomReader(
+      thingSchemaWithInline,
+      `from Thing access {
+         foo {a, d},
+         bar
+      }`,
+      'Thing',
+      'Thing {foo: inline Foo {a: Text, d: Text}, bar: Text}'
+    );
+    // List of inline entity tests
+    const thingSchemaWithInlineList = `
+      schema Thing
+        foo: List<inline Foo { a: Text, b: Text, c: Text, d: Text, e: Text }>
+        bar: Text
+    `;
+    await verifyPhantomReader(
+      thingSchemaWithInlineList,
+      `from Thing access {
+         foo {a, d, e},
+      }`,
+      'Thing',
+      'Thing {foo: List<inline Foo {a: Text, d: Text, e: Text}>}'
+    );
+    await verifyPhantomReader(
+      thingSchemaWithInlineList,
+      `from Thing access {
+         foo {a, d},
+         bar
+      }`,
+      'Thing',
+      'Thing {foo: List<inline Foo {a: Text, d: Text}>, bar: Text}'
+    );
+    const thingSchemaWithTuples = `
+      schema Thing
+        a: (Text, Number)
+        b: [(Text, Number)]
+        c: List<(Text, Number)>
+        d: (Text, Number)
+        e: (Text, Number)
+        f: Text
+    `;
+    await verifyPhantomReader(
+      thingSchemaWithTuples,
+      'from Thing access { f }',
+      'Thing',
+      'Thing {f: Text}'
+    );
+    // TODO(b/174501386): Tuples are not supported in the policy language yet.
+    await assertThrowsAsync(
+      async () => verifyPhantomReader(
+        thingSchemaWithTuples,
+        'from Thing access { a }',
+        'Thing',
+        'Thing {a: (Text, Number)}'
+      ),
+      FieldPathError,
+      `Unsupported field type`
+    );
+  });
```

Comment:
could you add a test case for the error case - `Unable to find max read type for handle`?
maybe also a case when the writer fields' and the policy fields intersection is empty?

Reply:
Done. I added the following tests:
 - `Unable to find max read type` error.
 - example where intersection of writer fields and policy fields is empty.